src/proxy/NearCachedMapProxy.ts

```diff
@@ -138,28 +138,28 @@ export class NearCachedMapProxy<K, V> extends MapProxy<K, V> {
                                   ttl: number | Long = -1,
                                   maxIdle?: number | Long): Promise<V> {
         return super.putIfAbsentInternal(keyData, valueData, ttl, maxIdle)
-            .then<V>(this.invalidateCacheEntryAndReturn.bind(this, keyData));
+            .then<V>(oldValue => this.invalidateCacheEntryAndReturn(keyData, oldValue));
     }
 
     protected putTransientInternal(keyData: Data,
                                    valueData: Data,
                                    ttl: number | Long = -1,
                                    maxIdle?: number | Long): Promise<void> {
         return super.putTransientInternal(keyData, valueData, ttl, maxIdle)
-            .then<void>(this.invalidateCacheEntryAndReturn.bind(this, keyData));
+            .then<void>(() => this.invalidateCacheEntryAndReturn(keyData));
     }
 
     protected executeOnKeyInternal(keyData: Data, proData: Data): Promise<V> {
         return super.executeOnKeyInternal(keyData, proData)
-            .then<V>(this.invalidateCacheEntryAndReturn.bind(this, keyData));
+            .then<V>(oldValue => this.invalidateCacheEntryAndReturn(keyData, oldValue));
```

Comment:
Maybe we can replace this with `result` as this is the result of the entry processor